WordPress/gutenberg
# DRY up ContentBlocksList and BlockInspectorLockedBlocks

## What?
Follows https://github.com/WordPress/gutenberg/pull/50857.

Extract a reusable component out of `ContentBlocksList` and `BlockInspectorLockedBlocks` so that we're not repeating the same code that outputs a list of block navigation links.

## Why?
Less code good.

## Testing Instructions
1. Go to Appearance → Editor → Pages and select a page.
1. The right sidebar should display a list containing Post Featured Image, Title, and Post Content. Clicking on items in this list should select the block. The items should appear selected if the block or one of its children are selected.
1. Create a new post and insert a pattern that contains a `contentOnly` lock. ([Here's one.](https://gist.github.com/richtabor/ddeea41ced691721318649bea8ce9db8))
1. Select the pattern.
1. The right sidebar should display a list containing the editable blocks within that pattern. Clicking on items in this list should select the block. The items should appear selected if the block is selected.
